tests/subsys/emul/testcase.yaml

```diff
@@ -0,0 +1,6 @@
+tests:
+  subsys.emul:
+    tags: devicetree
+    # We only need this to run on one platform so use native_posix as it
+    # will mostly likely be the fastest.
```

Comment:
If I'm not wrong, native_posix is the only supported platform when using emulators (because you really need the I2C emulator), so this comment is misleading.

Reply:
> If I'm not wrong, native_posix is the only supported platform when using emulators (because you really need the I2C emulator), so this comment is misleading.

Actually emulators can be used with Qemu as well. I've got a PR showing this that adds a new sbs_gauge emulator.

Reply:
so then platform_allow should reflect this capability as well. You can choose native_posix as an integration_platforms.

Reply:
My intent here was to avoid redundant testing. Since the feature under test doesn't actually do anything at runtime and only relies on the compiler, linker, and gen_defines.py, I figured testing on one platform would be as good as testing on all of them. But if you want this to run on anything that it can run on, I'm OK with that.

I'm still trying to figure out the best way to do that. My `app.overlay` references `i2c0`, which is a node in `native_posix.dts`, but I found ~400 .dts files in which it is absent. Consequently, I get devicetree errors during configuration for several of the qemu platforms if I remove this `platform_allow`. How is this typically handled?

Reply:
Relying on a hardcoded node isn't a good idea for tests/samples that are meant to run on multiple platforms. A couple of things you could do:

- Filter based on dt, e.g. `filter: dt_compat_enabled("zephyr,i2c-emul-controller")`
- Use an alias for the I2C emulator, e.g. `i2c-emul = &i2c0;`.

Reply:
If it's really a test that you want as a subsys kind of unit test, then it's ok to restrict to native_posix, but then comment needs to be deleted or adjusted to avoid confusions.

Reply:
This test should use platform_allow: native_posix

Reply:
Per OOB discussion with @yperess, I added `depends_on: i2c` and `filter: CONFIG_EMUL`. Currently, this limits the test to `native_posix`. If we support emulated I2C on other platforms in the future, it will run on those too.

Reply:
Addendum: Apparently, CI runs more platforms than my local twister runs, including some that support I2C and emulators but have different I2C layouts than `native_posix`. For instance, `swan_r5` numbers its busses starting at 1 instead of 0. Coming up with usable devicetree overlays for each such board seems like it would just make this test pointlessly voluminous. I'm OK with `platform_allow: native_posix` if you are.